### PR TITLE
Exposing MarshalByRefObject

### PR DESCRIPTION
**Pull request by ramarag**
See https://github.com/dotnet/corefx/issues/11809#issuecomment-252778257

@jkotas @weshaggard @stephentoub  PTAL

### CODE REVIEW

**Comment by jkotas** [comment]
While you are on it, could you please also delete about the dozen ifdefs for MarshalByRefObject base-class, like:

```
#if FEATURE_REMOTING
    public abstract class Stream : MarshalByRefObject, IDisposable {
#else // FEATURE_REMOTING
    public abstract class Stream : IDisposable {
#endif // FEATURE_REMOTING
```

should be just:
```
    public abstract class Stream : MarshalByRefObject, IDisposable {
```

**Comment by jkotas** [comment]
LGTM otherwise.

File: src/mscorlib/src/System/MarshalByRefObject.cs [hunk 1]
```diff
@@ -35,10 +35,10 @@ namespace System {
 [System.Runtime.InteropServices.ComVisible(true)]
     public abstract class MarshalByRefObject 
     {
-#if FEATURE_REMOTING    
         private Object __identity;
```
Comment (jkotas) [inline src/mscorlib/src/System/MarshalByRefObject.cs]: You either need to keep this under `FEATURE_REMOTING` ifdef; or remove the matching ifdefs in VM. I think that keeping this under `FEATURE_REMOTING` ifdef is better.

The test failures are caused by this problem.

File: src/mscorlib/src/System/MarshalByRefObject.cs [hunk 2]
```diff
@@ -256,7 +244,28 @@ internal static bool CanCastToXmlTypeHelper(RuntimeType castType, MarshalByRefOb
 
             return o.CanCastToXmlType(xmlTypeName, xmlTypeNamespace);
         } // CanCastToXmlType
-
+#else
+        protected MarshalByRefObject() { }
```
Comment (weshaggard) [inline src/mscorlib/src/System/MarshalByRefObject.cs]: What about CreateObjRef? https://github.com/dotnet/standard/blob/master/netstandard/ref/mscorlib.cs#L2449.

Comment (weshaggard) [inline src/mscorlib/src/System/MarshalByRefObject.cs]: Thanks for pointing out the discussion (I'm way far behind on my github notifications), I just added a comment there that we should keep this member and it's closure in System.Runtime and thus implement them in System.Private.CoreLib.

**Comment by jkotas** [comment]
@dotnet-bot test this please

**Comment by ramarag** [comment]
@jkotas @weshaggard do we have a decsion whether we are CreateObjRef to marshalbyrefobject class
and where objectref and associated interfaces should be implemented ?

**Comment by ramarag** [comment]
@jkotas @weshaggard do we have a decision whether we are going to add CreateObjRef to marshalbyrefobject class
and where objectref and associated interfaces should be implemented ?

**Comment by jkotas** [comment]
@weshaggard removed CreateObjRef from netstandard (https://github.com/dotnet/standard/commit/80799d92215df2fa0751c4a632def7379b0c5b3d), so CreateObjRef is not needed anymore.

**Comment by ramarag** [comment]
does this mean objectref and associated interfaces need not be implemented for netstandard2 ?

**Comment by jkotas** [comment]
Yes.

**Comment by weshaggard** [comment]
@ramarag correct see the conversation at https://github.com/dotnet/corefx/issues/11809#issuecomment-252786425

**Pull request merged**